### ISSUE TITLE
`dotnet format --fix-style` reports errors that `dotnet build` does not

### ISSUE THREAD

**Issue by bradwilson** [dotnet/format#971]
1. Clone https://github.com/xunit/xunit
2. Edit `.config/dotnet-tools.json` and update `dotnet-format` to `5.0.211103`
3. Run `dotnet tool restore`

Run a normal build: `dotnet build`. Note there are no build warnings or errors:

```
[...]
  xunit.v3.runner.tdnet.tests -> C:\Dev\xunit\xunit\src\xunit.v3.runner.tdnet.tests\bin\Debug\net472\xunit.v3.runner.tdnet.tests.exe
  xunit.v3.common.tests -> C:\Dev\xunit\xunit\src\xunit.v3.common.tests\bin\Debug\netcoreapp2.1\xunit.v3.common.tests.dll
  xunit.v3.runner.inproc.console.tests -> C:\Dev\xunit\xunit\src\xunit.v3.runner.inproc.console.tests\bin\Debug\netcoreapp2.1\xunit.v3.runner.inproc.console.tests.dll
  xunit.v3.runner.utility.tests -> C:\Dev\xunit\xunit\src\xunit.v3.runner.utility.tests\bin\Debug\netcoreapp2.1\xunit.v3.runner.utility.tests.dll
  xunit.v3.assert.tests -> C:\Dev\xunit\xunit\src\xunit.v3.assert.tests\bin\Debug\netcoreapp2.1\xunit.v3.assert.tests.dll
  xunit.v3.runner.msbuild.tests -> C:\Dev\xunit\xunit\src\xunit.v3.runner.msbuild.tests\bin\Debug\net472\xunit.v3.runner.msbuild.tests.exe

Build succeeded.
    0 Warning(s)
    0 Error(s)
```

Now run `dotnet format --fix-style`. It appears to be reporting nullability errors that a normal build does not:

```
  Formatting code files in workspace 'C:\Dev\xunit\xunit\xunit.sln'.
  Warnings were encountered while loading the workspace. Set the verbosity option to the 'diagnostic' level to log warnings.
  src\xunit.v3.assert\Asserts\Sdk\AssertEqualityComparerAdapter.cs(38,32): Converting null literal or possible null value to non-nullable type. (CS8600)
  src\xunit.v3.assert\Asserts\Sdk\AssertEqualityComparerAdapter.cs(38,38): Converting null literal or possible null value to non-nullable type. (CS8600)
  src\xunit.v3.assert\Asserts\CollectionAsserts.cs(285,46): Converting null literal or possible null value to non-nullable type. (CS8600)
  src\xunit.v3.assert\Asserts\CollectionAsserts.cs(309,46): Converting null literal or possible null value to non-nullable type. (CS8600)
  src\xunit.v3.assert\Asserts\CollectionAsserts.cs(588,15): Converting null literal or possible null value to non-nullable type. (CS8600)
  src\xunit.v3.common\Extensions\XunitSerializationInfoExtensions.cs(18,4): Converting null literal or possible null value to non-nullable type. (CS8600)
  src\xunit.v3.core\Sdk\ExceptionAggregator.cs(134,74): Possible null reference assignment. (CS8601)
  src\xunit.v3.core\Sdk\v3\Utility\TestCollectionFactoryHelper.cs(29,15): Dereference of a possibly null reference. (CS8602)
  src\xunit.v3.assert\Asserts\Sdk\AssertEqualityComparerAdapter.cs(38,32): Converting null literal or possible null value to non-nullable type. (CS8600)
  src\xunit.v3.assert\Asserts\Sdk\AssertEqualityComparerAdapter.cs(38,38): Converting null literal or possible null value to non-nullable type. (CS8600)
  src\xunit.v3.runner.common\Frameworks\_TestFrameworkOptions.cs(91,17): Nullability of reference types in return type of 'TValue _TestFrameworkOptions.GetValue<TValue>(string name)' doesn't match implicitly implemented member 'TValue _ITestFrameworkOptions.GetValue<TValue>(string name)' (possibly because of nullability attributes). (CS8766)
  Unable to fix CS8766. No associated code fix found.
  Unable to fix CS8601. No associated code fix found.
  Unable to fix CS8602. No associated code fix found.
  Formatted code file 'C:\Dev\xunit\xunit\src\xunit.v3.assert\Asserts\Sdk\AssertEqualityComparerAdapter.cs'.
  Formatted code file 'C:\Dev\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs'.
  Formatted code file 'C:\Dev\xunit\xunit\src\xunit.v3.common\Extensions\XunitSerializationInfoExtensions.cs'.
  Formatted code file 'C:\Dev\xunit\xunit\src\xunit.v3.core\Sdk\ExceptionAggregator.cs'.
  Formatted code file 'C:\Dev\xunit\xunit\src\xunit.v3.core\Sdk\v3\Utility\TestCollectionFactoryHelper.cs'.
  Formatted code file 'C:\Dev\xunit\xunit\src\xunit.v3.assert\Asserts\Sdk\AssertEqualityComparerAdapter.cs'.
  Formatted code file 'C:\Dev\xunit\xunit\src\xunit.v3.runner.common\Frameworks\_TestFrameworkOptions.cs'.
  Format complete in 17066ms.
```

Note that since xUnit.net targets `netstandard2.0`, I am using [`TunnelVisionLabs.ReferenceAssemblyAnnotator`](https://github.com/tunnelvisionlabs/ReferenceAssemblyAnnotator) to get nullability annotations. That pulls its annotations from [`Microsoft.NETCore.App.Ref`](https://www.nuget.org/packages/Microsoft.NETCore.App.Ref). It appears that whatever `dotnet format` is doing is different than what `dotnet build` is doing.

(There may be a secondary issue here, which is: none of the claimed changes to the listed files are actually committed to disk. Not sure if that's because there are a few "unfixable" issues, but it does feel like `dotnet format` should be willing to commit the changes that it _did_ make.)

**Comment by jmarolf** [comment]
I do see these in the latest version of the C# compiler with `dotnet build`

```
> dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.103
 Commit:    72dec52dbd

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.103\
```

```
> dotnet build
Microsoft (R) Build Engine version 16.8.3+39993bd9d for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  xunit.v3.assert.NonNullable -> C:\source\xunit\xunit\src\xunit.v3.assert.NonNullable\bin\Debug\netstandard2.0\xunit.v3.assert.NonNullable.dll
  xunit.v1.tests -> C:\source\xunit\xunit\src\xunit.v1.tests\bin\Debug\net45\xunit.v1.tests.dll
C:\source\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs(480,11): error CS8603: Possible null reference return. [C:\source\xunit\xunit\src\xunit.v3.assert\xunit.v3.assert.csproj]
C:\source\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs(505,45): error CS8620: Argument of type 'Predicate<T>' cannot be used for parameter 'predicate' of type 'Predicate<T?>' in 'T? Assert.GetSingleResult<T?>(IEnumerable<T?> collection, Predicate<T?>? predicate, string? expectedArgument, out Exception? exceptionToThrow, T? defaultValue = default(T?))' due to differences in the nullability of reference types. [C:\source\xunit\xunit\src\xunit.v3.assert\xunit.v3.assert.csproj]
C:\source\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs(510,11): error CS8603: Possible null reference return. [C:\source\xunit\xunit\src\xunit.v3.assert\xunit.v3.assert.csproj]
C:\source\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs(540,11): error CS8603: Possible null reference return. [C:\source\xunit\xunit\src\xunit.v3.assert\xunit.v3.assert.csproj]
  xunit.v2.tests -> C:\source\xunit\xunit\src\xunit.v2.tests\bin\Debug\net452\xunit.v2.tests.dll
  xunit.v3.common -> C:\source\xunit\xunit\src\xunit.v3.common\bin\Debug\netstandard2.0\xunit.v3.common.dll
  xunit.v2.tests -> C:\source\xunit\xunit\src\xunit.v2.tests\bin\Debug\netcoreapp2.1\xunit.v2.tests.dll
  xunit.v3.runner.common -> C:\source\xunit\xunit\src\xunit.v3.runner.common\bin\Debug\net472\xunit.v3.runner.common.net472.dll
  xunit.v3.core -> C:\source\xunit\xunit\src\xunit.v3.core\bin\Debug\netstandard2.0\xunit.v3.core.dll
  xunit.v3.runner.common -> C:\source\xunit\xunit\src\xunit.v3.runner.common\bin\Debug\netstandard2.0\xunit.v3.runner.common.netstandard20.dll
  xunit.v3.runner.utility -> C:\source\xunit\xunit\src\xunit.v3.runner.utility\bin\Debug\net472\xunit.v3.runner.utility.net472.dll
  xunit.v3.runner.inproc.console -> C:\source\xunit\xunit\src\xunit.v3.runner.inproc.console\bin\Debug\net472\xunit.v3.runner.inproc.console.dll
  xunit.v3.runner.utility -> C:\source\xunit\xunit\src\xunit.v3.runner.utility\bin\Debug\netstandard2.0\xunit.v3.runner.utility.netstandard20.dll
  xunit.v3.runner.inproc.console -> C:\source\xunit\xunit\src\xunit.v3.runner.inproc.console\bin\Debug\netcoreapp2.1\xunit.v3.runner.inproc.console.dll
  xunit.v3.runner.tdnet -> C:\source\xunit\xunit\src\xunit.v3.runner.tdnet\bin\Debug\net472\xunit.v3.runner.tdnet.dll
  xunit.v3.runner.msbuild -> C:\source\xunit\xunit\src\xunit.v3.runner.msbuild\bin\Debug\net472\xunit.v3.runner.msbuild.dll
  xunit.v3.runner.console -> C:\source\xunit\xunit\src\xunit.v3.runner.console\bin\Debug\net472\xunit.v3.runner.console.exe
  xunit.v3.runner.console -> C:\source\xunit\xunit\src\xunit.v3.runner.console\bin\Debug\net48\xunit.v3.runner.console.exe

Build FAILED.

C:\source\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs(480,11): error CS8603: Possible null reference return. [C:\source\xunit\xunit\src\xunit.v3.assert\xunit.v3.assert.csproj]
C:\source\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs(505,45): error CS8620: Argument of type 'Predicate<T>' cannot be used for parameter 'predicate' of type 'Predicate<T?>' in 'T? Assert.GetSingleResult<T?>(IEnumerable<T?> collection, Predicate<T?>? predicate, string? expectedArgument, out Exception? exceptionToThrow, T? defaultValue = default(T?))' due to differences in the nullability of reference types. [C:\source\xunit\xunit\src\xunit.v3.assert\xunit.v3.assert.csproj]
C:\source\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs(510,11): error CS8603: Possible null reference return. [C:\source\xunit\xunit\src\xunit.v3.assert\xunit.v3.assert.csproj]
C:\source\xunit\xunit\src\xunit.v3.assert\Asserts\CollectionAsserts.cs(540,11): error CS8603: Possible null reference return. [C:\source\xunit\xunit\src\xunit.v3.assert\xunit.v3.assert.csproj]
    0 Warning(s)
    4 Error(s)

Time Elapsed 00:00:09.06
```

**Comment by bradwilson** [comment]
FYI:

```
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.103
 Commit:    72dec52dbd

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.103\

Host (useful for support):
  Version: 5.0.3
  Commit:  c636bbdc8a

.NET SDKs installed:
  3.1.406 [C:\Program Files\dotnet\sdk]
  5.0.103 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by jmarolf** [comment]
Now that the 5.0.200 SDK is released I am going to close this. This compiler change has shipped in both VS (in VS 2019 16.9) and the SDK (5.0.200)
